nexus/tests/integration_tests/schema.rs

```diff
@@ -975,9 +977,11 @@ fn before_23_0_0(client: &Client) -> BoxFuture<'_, ()> {
         // no corresponding silo.
         client.batch_execute(&format!("INSERT INTO ip_pool
             (id, name, description, time_created, time_modified, time_deleted, rcgen, silo_id, is_default) VALUES
+          ('{POOL0}', 'pool2', '', now(), now(), now(), 1, '{SILO2}', true),
```

Comment:
With this deleted pool added to the test data but the **fix commented out**, the test fails in the way we expect, which matches the problem on dogfood: the (`pool3`, `silo2`) link gets `is_default=false` instead of `is_default=true`.